funcframework/framework.go

```diff
@@ -206,6 +217,42 @@ func wrapEventFunction(fn interface{}) (http.Handler, error) {
 	}), nil
 }
 
+func wrapTypedFunction(fn interface{}) (http.Handler, error) {
```

Comment:
For Go, we have to require error as the last return of the user's function. Otherwise, users have no way to express error cases except to panic, which isn't good practice in Go.

That means we'll have to be flexible and accept 1-2 returns, probably double checking the type of the last return type: https://github.com/GoogleCloudPlatform/functions-framework-go/blob/226db39d7dcdd7c7d02d33b620d9f897d84b5d2d/funcframework/events.go#L173-L177
Similar to how Event functions do it, we should return a 500 on the user's behalf if they return an error from their function: https://github.com/GoogleCloudPlatform/functions-framework-go/blob/226db39d7dcdd7c7d02d33b620d9f897d84b5d2d/funcframework/framework.go#L281-L284